### ISSUE TITLE
#if blocks result in code being moved around incorrectly

### ISSUE THREAD

**Issue by 7sharp9** [fsprojects/fantomas#282]
As an example from this code base `TokenMatcher.fs`:

```fsharp
module internal Fantomas.TokenMatcher

open System
open System.Collections.Generic
open System.Diagnostics
open Fantomas
open Microsoft.FSharp.Compiler.Range
open Microsoft.FSharp.Compiler.PrettyNaming
open Microsoft.FSharp.Compiler.SourceCodeServices
open System.Text.RegularExpressions

#if INTERACTIVE
type Debug = Console
#endif

type Token = 
   | EOL
   | Tok of FSharpTokenInfo * int
   override x.ToString() =
        match x with
        | EOL -> "<EOL>"
        | Tok(tokInfo, l) ->
            sprintf "Tok(%O, %O)" tokInfo.TokenName l
```

This will get formatted to:

```fsharp
module internal Fantomas.TokenMatcher

open System
open System.Collections.Generic
open System.Diagnostics
open Fantomas
open Microsoft.FSharp.Compiler.Range
open Microsoft.FSharp.Compiler.PrettyNaming
open Microsoft.FSharp.Compiler.SourceCodeServices
open System.Text.RegularExpressions
#if INTERACTIVE

type Token =
    
#endif
| EOL
    | Tok of FSharpTokenInfo * int
    override x.ToString() =
        match x with
        | EOL -> "<EOL>"
        | Tok(tokInfo, l) -> sprintf "Tok(%O, %O)" tokInfo.TokenName l
```

`type Debug = Console` is erased, `type Token =` is moved into the #if block.

**Comment by jindraivanek** [comment]
This is now formatted to:
```
module internal Fantomas.TokenMatcher

open System
open System.Collections.Generic
open System.Diagnostics
open Fantomas
open Microsoft.FSharp.Compiler.Range
open Microsoft.FSharp.Compiler.PrettyNaming
open Microsoft.FSharp.Compiler.SourceCodeServices
open System.Text.RegularExpressions
#if INTERACTIVE

type Debug = Console
#endif


type Token =
    | EOL
    | Tok of FSharpTokenInfo * int
    override x.ToString() =
        match x with
        | EOL -> "<EOL>"
        | Tok(tokInfo, l) -> sprintf "Tok(%O, %O)" tokInfo.TokenName l
```

so only remaining problem is added/removed newlines. Probably bug in TokenMatcher.

**Comment by nojaf** [comment]
This issue is now solved by [trivia](#434).
